SDWebImage/UIView+WebCache.m

```diff
@@ -224,6 +221,15 @@ - (void)sd_setImage:(UIImage *)image imageData:(NSData *)imageData basedOnClassO
         };
     }
 #endif
+    else {
+        finalSetImageBlock = ^(UIImage *setImage, NSData *setImageData){
+#if SD_MAC
+            view.layer.contents = setImage;
```

Comment:
This is not suitable code. You can not assume all View are CALayer based, and it's suitable to touch `contents` for rendering.

For example, on macOS, there are CPU-based view && layer-backed view && layer-hosting view. The layer may be nil.
For example, on iOS, there are many layer like `CAEAGLLayer`, `CATextLayer `, which the `contents` is always be nil and been ignored. You should not touch them.
For customization, always pass your draw code inside that `setImageBlock`.

Reply:
Some code like below
Gif downloaded and run it again, gif will be freezed.

```objc
#import "ViewController.h"
#import <SDWebImage/SDWebImage.h>

@interface ViewController () {
    FLAnimatedImageView *_imageView;
}

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.

    _imageView = [[FLAnimatedImageView alloc] initWithFrame:CGRectMake(50, 100, 200, 200)];
    _imageView.contentMode = UIViewContentModeScaleAspectFit;
    _imageView.backgroundColor = UIColor.lightGrayColor;
    [self.view addSubview:_imageView];
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];

    NSURL *url = [NSURL URLWithString:@"https://github.com/cntrump/pics/raw/master/SDWebImage/giphy.gif"];
    UIImage *cacheImage = [[SDWebImageManager sharedManager].imageCache imageFromCacheForKey:url.absoluteString];

    [_imageView sd_setImageWithURL:url placeholderImage:cacheImage options:SDWebImageRetryFailed completed:^(UIImage * _Nullable image, NSError * _Nullable error, SDImageCacheType cacheType, NSURL * _Nullable imageURL) {

    }];
}

@end

```

Reply:
This bug triggered when placeholder image is a GIF same as the url.
Actually, I just want display a poster like this:

First time, cache the image as poster

```objc
    NSURL *url = [NSURL URLWithString:@"https://github.com/cntrump/pics/raw/master/SDWebImage/giphy.gif"];

    [_imageView sd_setImageWithURL:url placeholderImage:_posterImage options:SDWebImageRetryFailed completed:^(UIImage * _Nullable image, NSError * _Nullable error, SDImageCacheType cacheType, NSURL * _Nullable imageURL) {
        self->_posterImage = image;
    }];
```

Second time, using the poster as placeholder

```objc
    NSURL *url = [NSURL URLWithString:@"https://github.com/cntrump/pics/raw/master/SDWebImage/giphy.gif"];

    [_imageView sd_setImageWithURL:url placeholderImage:_posterImage options:SDWebImageRetryFailed completed:^(UIImage * _Nullable image, NSError * _Nullable error, SDImageCacheType cacheType, NSURL * _Nullable imageURL) {
    }];
```

Reply:
My fault. We have to keep placeholder check, but need a better way to filter the cases to distringuish the difference between `user provided placeholder image instance` and `image queried from cache instance`. Because placeholder should not trigger any `dispatch queue` async operation, or the Cell-Reusing will be incorrect...I'll try to do this.

Reply:
OK, thanks.